lib/Emitter.php

```diff
@@ -25,32 +26,32 @@
             /**
              * @var callable
              */
-            private $resolve;
+            private $complete;
 
             /**
              * @var callable
              */
             private $fail;
 
             public function __construct() {
-                $this->stream = new Internal\PrivateStream(
-                    function (callable $emit, callable $resolve, callable $fail) {
+                $this->iterator = new Internal\PrivateIterator(
+                    function (callable $emit, callable $complete, callable $fail) {
                         $this->emit = $emit;
-                        $this->resolve = $resolve;
+                        $this->complete = $complete;
                         $this->fail = $fail;
                     }
                 );
             }
 
             /**
-             * @return \Amp\Stream
+             * @return \Amp\Iterator
              */
-            public function stream(): Stream {
-                return $this->stream;
+            public function getIterator(): Iterator {
```

Comment:
Well, I don't buy your argument - I don't really see how this will make it easier to distinguish that.

Additionally, technically, `iterate()` could also each time return an other object. Promise and Iterator have both their state managed by another entity.

I'll go with my OCD and prefer symmetry here.